horace_core/GUI/horace.m

```diff
@@ -1436,52 +1455,32 @@ function Cut_do_cut_pushbutton_Callback(hObject, eventdata, handles)
 else
     try
         %must strip out square brackets, if user has inserted them:
-        s1=strfind(a1,'['); s2=strfind(a1,']');
-        if isempty(s1) && isempty(s2)
-            a1new=textscan(a1,'%f','delimiter',',');
-        elseif ~isempty(s1) && ~isempty(s2)
-            a1=a1(s1+1:s2-1);
-            a1new=textscan(a1,'%f','delimiter',',');
-        else
+        [a1new,ok] = read_vector(a1);
```

Comment:
Using proper error handling (since all the error messages are the same, this could be reduced to:
```matlab
try
    a1new = read_vector(a1);
    a2new = read_vector(a2);
    a3new = read_vector(a3);
    a4new = read_vector(a4);
catch ME
    switch ME.identifier
      case HORACE:GUI:vec_read_error
          mess1='   Ensure binning values are entered if the form of lo,step,hi / step / lo,hi    ';
          mess2='NB: enter 0 if you wish to use intrinsic binning and entire data range along axis';
          set(handles.message_info_text,'String',char({mess_initialise,mess1,mess2}));
          guidata(gcbo,handles);
          return;
      otherwise
          rethrow(ME)
    end
end
```

Reply:
See above about the exceptions, but logic of this change is  wrong -- the checks are for incorrect value provided by user. In some cases empty is correct value, in some they are not. In other cases you read bunch or [] separated values. 

Not saying that the code in current `horace` is not entirely correct.  It should inform user about what exactly box is actually wrong. 

There are bunch of important changes asked to be done to GUI. Its not the time to start them now -- it would not stop easy.

Reply:
Empty would not return an error according to your definition of `read_vector`, I don't see your point there. 

Doing minimal changes now while we're there that make later cleanup easier just make sense to me.